samples/custom_dispatch/cpu/plugin/mlp_spec.mlir

```diff
@@ -0,0 +1,142 @@
+          %c0 = arith.constant 0 : index
+          %m = arith.index_cast %m_i32 : i32 to index
+          %n = arith.index_cast %n_i32 : i32 to index
+          %k = arith.index_cast %k_i32 : i32 to index
+          %lhs = hal.interface.binding.subspan set(0) binding(0) type(storage_buffer) alignment(64) offset(%c0) : memref<?x?xf32>{%m, %k}
+          %rhs = hal.interface.binding.subspan set(0) binding(1) type(storage_buffer) alignment(64) offset(%c0) : memref<?x?xf32>{%k, %n}
+          %result = hal.interface.binding.subspan set(0) binding(2) type(storage_buffer) alignment(64) offset(%c0) : memref<?x?xf32>{%m, %n}
+          func.call @mlp_external(%lhs, %rhs, %result, %m_i32, %n_i32, %k_i32) : (memref<?x?xf32>, memref<?x?xf32>, memref<?x?xf32>, i32, i32, i32) -> ()
+          return
+        }
+      }
+    }
+  }
+
+  func.func private @call_mlp(%lhs : tensor<?x?xf32>, %rhs : tensor<?x?xf32>, %init1 : tensor<?x?xf32>, %init2 : tensor<?x?xf32>) -> tensor<?x?xf32> {
+    %c0 = arith.constant 0 : index
+    %c1 = arith.constant 1 : index
+    %m = tensor.dim %lhs, %c0 : tensor<?x?xf32>
+    %n = tensor.dim %rhs, %c1 : tensor<?x?xf32>
+    %k = tensor.dim %lhs, %c1 : tensor<?x?xf32>
+    %m_i32 = arith.index_cast %m : index to i32
+    %n_i32 = arith.index_cast %n : index to i32
+    %k_i32 = arith.index_cast %k : index to i32
+
+    %mlp_result = flow.dispatch @executable::@x86_64::@mlp[](%lhs, %rhs, %m_i32, %n_i32, %k_i32) {
+      hal.interface.bindings = [
+        #hal.interface.binding<0, 0>,
+        #hal.interface.binding<0, 1>,
+        #hal.interface.binding<0, 2>
+      ],
+      // HACK: keep the executable live through DCE. Only required when
+      // using the automatic variant selection.
+      hal.executable.ref = [@executable]
+    } : (tensor<?x?xf32>{%m, %k}, tensor<?x?xf32>{%k, %n}, i32, i32, i32) -> tensor<?x?xf32>{%m, %n}  
+    return %mlp_result : tensor<?x?xf32>    
+  }
+
+  transform.named_sequence @match_mlp(%root: !transform.any_op {transform.readonly}) -> (!transform.any_value, !transform.any_value) {
+    %ins, %outs = transform.iree.match.cast_compatible_dag_from_root %root {
+      ^bb0(%lhs: tensor<?x?xf32>, %rhs: tensor<?x?xf32>, %init1 : tensor<?x?xf32>, %init2 : tensor<?x?xf32>):
+        %cst = arith.constant 0.0 : f32
+        %fill = linalg.fill ins(%cst : f32) outs(%init1 : tensor<?x?xf32>) -> tensor<?x?xf32>
+        %matmul = linalg.matmul
+            ins(%lhs, %rhs : tensor<?x?xf32>, tensor<?x?xf32>)
+                outs(%fill : tensor<?x?xf32>) -> tensor<?x?xf32>
+        %relu = linalg.generic {
+            indexing_maps = [affine_map<(d0, d1) -> (d0, d1)>,
+                             affine_map<(d0, d1) -> (d0, d1)>],
+            iterator_types = ["parallel", "parallel"]}
+            ins(%matmul : tensor<?x?xf32>)
+            outs(%init2 : tensor<?x?xf32>) {
+          ^bb0(%b0 : f32, %b1 : f32):
+            %0 = arith.maximumf %b0, %cst : f32
```

Comment:
Does this properly match the `%cst` value? i.e. if it was instead ` %cst = arith.constant 1.0`. I remember making the matching this op does just walk the body from the root, so ops without explicit use-def relations won't be visited.

Reply:
Yeah, this was a problem. I had to actually refactor the matching to make this not an issue. PTAL.